code/modules/mob/living/basic/ruin_defender/mimic/mimic.dm

```diff
@@ -0,0 +1,381 @@
+	return TRUE
+
+/mob/living/basic/mimic/crate/adjust_health(amount, updating_health = TRUE, forced = FALSE)
+	if(amount > 0)
+		trigger()
+	return ..()
+
+/mob/living/basic/mimic/crate/death()
+	var/obj/structure/closet/crate/C = new(get_turf(src))
+	// Put loot in crate
+	for(var/obj/O in src)
+		O.forceMove(C)
+	return ..()
+
+/mob/living/basic/mimic/crate/early_melee_attack(atom/target, list/modifiers, ignore_cooldown)
+	if(target == src)
+		toggle_open()
+		return FALSE
+	return ..()
+
+/mob/living/basic/mimic/crate/CanAllowThrough(atom/movable/mover, border_dir)
+	. = ..()
+	if(istype(mover, /obj/structure/closet))
+		return FALSE
+/**
+* Used to open and close the mimic
+*
+* Will insert tile contents into the mimic when closing
+* Will dump mimic contents into the time when opening
+* Does nothing if the mimic locked itself
+*/
+/mob/living/basic/mimic/crate/proc/toggle_open(mob/user)
+	if(locked)
+		if(user)
+			balloon_alert(user, "too stiff!")
+		return
+	if(!opened)
+		ADD_TRAIT(src, TRAIT_UNDENSE, MIMIC_TRAIT)
+		opened = TRUE
+		icon_state = "crateopen"
+		playsound(src, 'sound/machines/crate/crate_open.ogg', 50, TRUE)
+		for(var/atom/movable/movable as anything in src)
+			movable.forceMove(loc)
+	else
+		REMOVE_TRAIT(src, TRAIT_UNDENSE, MIMIC_TRAIT)
+		opened = FALSE
+		icon_state = "crate"
+		playsound(src, 'sound/machines/crate/crate_close.ogg', 50, TRUE)
+		for(var/atom/movable/movable as anything in get_turf(src))
+			if(movable != src && insert(movable) == -1)
+				playsound(src, 'sound/items/trayhit/trayhit2.ogg', 50, TRUE)
+				break
+/**
+* Called by toggle_open to put items inside the mimic when it's being closed
+*
+* Will return -1 if the insertion fails due to the storage capacity of the mimic having been reached
+* Will return FALSE if insertion fails
+* Will return TRUE if insertion succeeds
+* Arguments:
+* * AM - item to be inserted
+*/
+/mob/living/basic/mimic/crate/proc/insert(atom/movable/movable)
+	if(contents.len >= storage_capacity)
+		return -1
+	if(insertion_allowed(movable))
+		movable.forceMove(src)
+		return TRUE
+	return FALSE
+
+/mob/living/basic/mimic/crate/proc/insertion_allowed(atom/movable/movable)
+	if(ismob(movable))
+		if(!isliving(movable))  //Don't let ghosts and such get trapped in the beast.
+			return FALSE
+		var/mob/living/living = movable
+		if(living.anchored || living.buckled || living.incorporeal_move || living.has_buckled_mobs())
+			return FALSE
+		if(living.mob_size > MOB_SIZE_TINY) // Tiny mobs are treated as items.
+			if(living.density || living.mob_size > max_mob_size)
+				return FALSE
+			var/mobs_stored = 0
+			for(var/mob/living/living_mob in contents)
+				mobs_stored++
+				if(mobs_stored >= mob_storage_capacity)
+					return FALSE
+		living.stop_pulling()
+
+	else if(istype(movable, /obj/structure/closet))
+		return FALSE
+	else if(isobj(movable))
+		if(movable.anchored || movable.has_buckled_mobs())
```

Comment:
Might be worth checking for anchored right at the top of the proc to catch edge cases like anchored mobs or unexpectedly anchored items.